pop/service/service.go

```diff
@@ -0,0 +1,258 @@
+	// are correctly handled.
+	*onet.ServiceProcessor
+	path string
+	data *saveData
+	// channel to return the configreply
+	ccChannel chan *CheckConfigReply
+}
+
+type saveData struct {
+	// Pin holds the randomly chosen pin
+	Pin string
+	// Public key of linked pop
+	Public abstract.Point
+	// The final statement
+	Final *FinalStatement
+}
+
+// PinRequest prints out a pin if none is given, else it verifies it has the
+// correct pin, and if so, it stores the public key as reference.
+func (s *Service) PinRequest(req *PinRequest) (network.Message, onet.ClientError) {
+	if req.Pin == "" {
+		rand.Seed(int64(time.Now().Nanosecond()))
+		s.data.Pin = fmt.Sprintf("%06d", rand.Intn(1000000))
+		log.Info("PIN:", s.data.Pin)
+		return nil, onet.NewClientErrorCode(ErrorWrongPIN, "Read PIN in server-log")
+	}
+	if req.Pin != s.data.Pin {
+		return nil, onet.NewClientErrorCode(ErrorWrongPIN, "Wrong PIN")
+	}
+	s.data.Public = req.Public
+	s.save()
+	log.Lvl1("Successfully registered PIN/Public", s.data.Pin, req.Public)
+	return nil, nil
+}
+
+// StoreConfig saves the pop-config locally
+func (s *Service) StoreConfig(req *StoreConfig) (network.Message, onet.ClientError) {
+	log.Lvlf3("%s %v %x", s.Context.ServerIdentity(), req.Desc, req.Desc.Hash())
+	if req.Desc.Roster == nil {
+		return nil, onet.NewClientErrorCode(ErrorInternal, "no roster set")
+	}
+	if s.data.Public == nil {
+		return nil, onet.NewClientErrorCode(ErrorInternal, "Not linked yet")
+	}
+	s.data.Final = &FinalStatement{Desc: req.Desc, Signature: []byte{}}
+	s.save()
+	return &StoreConfigReply{req.Desc.Hash()}, nil
+}
+
+// FinalizeRequest returns the FinalStatement if all conodes already received
+// a PopDesc and signed off. The FinalStatement holds the updated PopDesc, the
+// pruned attendees-public-key-list and the collective signature.
+func (s *Service) FinalizeRequest(req *FinalizeRequest) (network.Message, onet.ClientError) {
+	log.Lvlf3("%s %+v", s.Context.ServerIdentity(), req)
+	if s.data.Public == nil {
+		return nil, onet.NewClientErrorCode(ErrorInternal, "Not linked yet")
+	}
+	if s.data.Final == nil || s.data.Final.Desc == nil {
+		return nil, onet.NewClientErrorCode(ErrorInternal, "No config found")
+	}
+	// Contact all other nodes and ask them if they already have a config.
+	s.data.Final.Attendees = make([]abstract.Point, len(req.Attendees))
+	copy(s.data.Final.Attendees, req.Attendees)
+	cc := &CheckConfig{s.data.Final.Desc.Hash(), req.Attendees}
+	for _, c := range s.data.Final.Desc.Roster.List {
+		if !c.ID.Equal(s.ServerIdentity().ID) {
+			log.Lvl3("Contacting", c, cc.Attendees)
+			err := s.SendRaw(c, cc)
+			if err != nil {
+				return nil, onet.NewClientErrorCode(ErrorInternal, err.Error())
+			}
+			rep := <-s.ccChannel
+			if rep == nil {
+				return nil, onet.NewClientErrorCode(ErrorOtherFinals,
+					"Not all other conodes finalized yet")
+			}
+		}
+	}
+	s.data.Final.Signature = []byte("signed")
```

Comment:
`crypto/sign/{VerifySchnorr,Schnorr}` is really easy now, you think it's worth putting there ? Or did you filed an issue for that ?